server/private-messages/database.ts

```diff
@@ -0,0 +1,148 @@
+/**
+ * Storage handling for offline PMs.
+ * By Mia.
+ * @author mia-pi-git
+ */
+import {SQL, FS} from '../../lib';
+import {findFriendship} from '../friends';
+import {EXPIRY_TIME, SEEN_EXPIRY_TIME, MAX_PENDING} from '.';
+import {Config} from '../config-loader';
+import {GlobalAuth, Auth} from '../user-groups';
+
+const globalAuth = new GlobalAuth(); // for blockpms
```

Comment:
It needs to be handled in the database process - the offline user's stored blockpms settings should (and is) be accessed in the transaction, since it requires a DB read.